portal/integrated_windows_authentication/src/main/java/com/esri/samples/integrated_windows_authentication/IntegratedWindowsAuthenticationController.java

```diff
@@ -0,0 +1,202 @@
+
+      // hide the progress indicator when the map is finished drawing
+      mapView.addDrawStatusChangedListener(drawStatusChangedEvent -> {
+        if (drawStatusChangedEvent.getDrawStatus() == DrawStatus.COMPLETED) {
+          progressIndicator.setVisible(false);
+        }
+      });
+
+    } catch (Exception e) {
+      // on any error, display the stack trace.
+      e.printStackTrace();
+    }
+  }
+
+  /**
+   * Handles searching the provided IWA secured portal.
+   */
+  @FXML
+  private void handleSearchPortalPress() {
+    // check for a url in the URL field
+    if (!portalUrlTextField.getText().isEmpty()) {
+      // keep hold of the portal we are searching and set a variable indicating that this is a secure portal, to allow retrieving portal items later
+      iwaSecuredPortal = new Portal(portalUrlTextField.getText(), true);
+
+      // clear any existing items in the list view
+      resultsListView.getItems().clear();
+
+      // clear the information about the previously loaded map
+      loadWebMapTextView.setText("");
+
+      // show portal load state
+      progressIndicator.setVisible(true);
+      loadStateTextView.setText("Searching for web map items on the portal at " + iwaSecuredPortal.getUri());
+
+      // load the portal items
+      iwaSecuredPortal.loadAsync();
+      iwaSecuredPortal.addDoneLoadingListener(() -> {
+        if (iwaSecuredPortal.getLoadStatus() == LoadStatus.LOADED) {
+          try {
+            // update load state in UI with the portal URI
+            loadStateTextView.setText("Connected to the portal on " + new URI(iwaSecuredPortal.getUri()).getHost());
+          } catch (URISyntaxException e) {
+            new Alert(Alert.AlertType.ERROR, "Error getting URI from portal: " + e.getMessage()).show();
+          }
+
+          // report the user name used for this connection
+          if (iwaSecuredPortal.getUser() != null) {
+            loadStateTextView.setText("Connected as: " + iwaSecuredPortal.getUser().getUsername());
+          } else {
+            // if connecting to an unsecured portal, no user credentials are needed to authenticate access
+            loadStateTextView.setText("Connected as: Anonymous");
+          }
+
+          // search the portal for web maps
+          ListenableFuture<PortalQueryResultSet<PortalItem>> portalItemResultFuture = iwaSecuredPortal.findItemsAsync(new PortalQueryParameters("type:(\"web map\" NOT \"web mapping application\")"));
+          portalItemResultFuture.addDoneListener(() -> {
+            try {
+              // get the result
+              PortalQueryResultSet<PortalItem> portalItemSet = portalItemResultFuture.get();
+              List<PortalItem> portalItems = portalItemSet.getResults();
+              // add the IDs and titles of the portal items to the list view
+              portalItems.forEach(portalItem -> resultsListView.getItems().add(new Pair<>(portalItem.getItemId(), portalItem.getTitle())));
```

Comment:
Make the listview of type `PortalItem` and use a string converter to display the title as the list item's text